src/components/alert/alert-component.ts

```diff
@@ -128,6 +140,22 @@ export class AlertCmp {
     }
   }
 
+  ionViewPreLoad() {
+    if (this.d.subview && this.d.subview.component) {
+
+      const componentFactory = this._cfr.resolveComponentFactory(this.d.subview.component);
+
+      const componentProviders = ReflectiveInjector.resolve([
+        { provide: NavParams, useValue: new NavParams(this.d.subview.args) }
+      ]);
+      const childInjector = ReflectiveInjector.fromResolvedProviders(componentProviders, this._viewport.parentInjector);
+
+      // ******** DOM WRITE ****************
+      const componentRef = this._viewport.createComponent(componentFactory, this._viewport.length, childInjector, []);
+      this.createdComponent = componentRef.instance;
```

Comment:
I also see a problem, if we don't change the "instance" of the viewcontroller, the lifecycle events will not be send to the user provided component. You should implement something like modal's case:
https://github.com/driftyco/ionic/blob/master/src/components/modal/modal-component.ts#L76-L81